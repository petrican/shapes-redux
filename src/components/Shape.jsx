import React from "react";

type Props = {
  id: string,
  shape: string,
  color: string
};

const Shape = (props: Props) => {
  let hexColor = "";
  let formedShape = "";

  switch (props.color) {
    case "red":
      hexColor = "#F40101";
      break;
    case "grey":
      hexColor = "#E1E1E1";
      break;
    case "green":
      hexColor = "#18C82B";
      break;
    case "blue":
      hexColor = "#2B9FFF";
      break;
    case "yellow":
      hexColor = "#F2BF44";
      break;
    case "purple":
      hexColor = "#890087";
      break;
    default:
    // do nothing
  }

  switch (props.shape) {
    case "triangle":
      formedShape = (
        <div
          className={`shape-triangle`}
          style={{ borderBottom: `125px solid ${hexColor}` }}
        ></div>
      );
      break;
    case "rectangle": // fix position for both rectangle and oval
    case "oval":
      formedShape = (
        <div
          className={`shape-${props.shape}`}
          style={{
            backgroundColor: `${hexColor}`,
            marginTop: "37px",
            marginBottom: "38px",
            marginLeft: "10px",
            marginRight: "15px"
          }}
        ></div>
      );
      break;
    default:
      formedShape = (
        <div
          className={`shape-${props.shape}`}
          style={{ backgroundColor: `${hexColor}` }}
        ></div>
      );
  }

  return <div className="shape-container">{formedShape}</div>;
};

export default Shape;
