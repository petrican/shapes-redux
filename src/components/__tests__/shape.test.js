'use strict';

import React from 'react';
import Shape from '../Shape';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Shape id="1" color="blue" shape="OVAL" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
