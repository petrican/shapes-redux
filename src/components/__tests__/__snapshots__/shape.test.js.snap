// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="shape-container"
>
  <div
    className="shape-OVAL"
    style={
      Object {
        "backgroundColor": "#2B9FFF",
      }
    }
  />
</div>
`;
